client/components/notice/announcement.js

```diff
@@ -0,0 +1,39 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+export class NoticeAnnouncement extends Component {
+	static defaultProps = {
+		spokenMessage: null,
+		status: null,
+		text: null,
+	};
+
+	static propTypes = {
+		spokenMessage: PropTypes.string,
+		status: PropTypes.oneOf( [ 'is-error', 'is-info', 'is-success', 'is-warning', 'is-plain' ] ),
+		text: PropTypes.oneOfType( [
+			PropTypes.arrayOf( PropTypes.oneOfType( [ PropTypes.string, PropTypes.node ] ) ),
+			PropTypes.oneOfType( [ PropTypes.string, PropTypes.node ] ),
+		] ),
+	};
+
+	constructor( props ) {
+		super( props );
+		this.container = document.getElementById( 'screen-reader-notice' );
```

Comment:
what should happen if that dom element doesn't exist or the `document` isn't available?

Reply:
also I think this could be a static class property, which would let you ditch the constructor entirely